infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -348,8 +348,9 @@ private void applyShadowStream(UUID streamId, long snapshot) {
             // OOM on applications running with a small memory footprint.  So for such tables, introduce an
             // additional limit of max number of entries(50 by default) applied in a single transaction.  This
             // algorithm is in line with the limits imposed in Compaction and Restore workflows.
-            if (bufferSize + smrEntry.getSerializedSize() > metadataManager.getRuntime().getParameters()
-                    .getMaxWriteSize() || maxEntriesLimitReached(streamId, buffer)) {
+            if (bufferSize + smrEntry.getSerializedSize() > replicationContext.getConfig(session).getMaxTransferSize()
```

Comment:
True, but we have to keep a buffer for this check also.  I had noticed in our tests that the final write size can increase beyond the runtime's limit despite having passed this check.  We can keep a lower buffer than the transfer size though.  Let me know what you think.

Reply:
I see...would the same 80% of uncompressed size work here as well?
Since the transfer size is configurable, there is no guarantee of having it as 80% of the size..

Reply:
Yes, I added a new variable for the apply size which considers just a fraction(85%) of the max uncompressed runtime size.  This will make it independent of the transfer size and allow us to write more data in a single transaction in the apply phase.